### PR TITLE
Transaction detail page ui bugfix

### PR DESCRIPTION
**Pull request by SerhiiShovkoplias**
## Description
transaction detail page design to pixel perfect. Created custom navigation bar for transaction detail controller. Part with emodji id is missed.

## Motivation and Context
It allows you to beautifully animate the transition to the new controller. In the old version, the standard nav bar appears before the controller leaves. It also allowed you to add the desired indent to the "Back" button from the left edge of the screen

tari-project/wallet-ios#169

## How Has This Been Tested?
Manual simulator testing

## Screenshots and/or video clip
Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-21 at 13 21 20
Simulator Screen Shot - iPhone Xs 13 4 - 2020-04-21 at 13 22 01
Simulator Screen Shot - iPhone 8 - 2020-04-21 at 13 23 44
Simulator Screen Shot - iPhone 8 Plus - 2020-04-21 at 13 25 21

## Types of changes

* [x] UI fix (non-breaking change which fixes a UI issue)
* [ ] Functionality bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I have tested this on multiple simulator devices with different screensizes.
* [x] I'm merging against the `development` branch.
* [x] Commits have been squashed into one commit with a descriptive commit message
* [x] I ran all tests before pushing.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added/changed tests to cover my changes if not UI changes.

### CODE REVIEW

File: MobileWallet/UIElements/EmoticonView.swift [hunk 1]
```diff
@@ -67,7 +67,24 @@ class EmoticonView: UIView {
     private var labelWidthConstraint: NSLayoutConstraint?
 
     private var emojiText: String!
-    private let RADIUS_POINTS: CGFloat = 6.0
+
+    private var RADIUS_POINTS: CGFloat = 6.0 {
```
Comment (Jasonvdb) [inline MobileWallet/UIElements/EmoticonView.swift:71]: Change to lower case here, now that it's not a const anymore. Kutsal and I decided to also just not use any capital naming conventions going forward so if you see any please refactor.

File: MobileWallet/UIElements/EmoticonView.swift [hunk 2]
```diff
@@ -67,7 +67,24 @@ class EmoticonView: UIView {
...
+        didSet (newValue) {
+            self.layer.cornerRadius = newValue
+            scrollView.layer.cornerRadius = newValue
+            label.layer.cornerRadius = newValue
+        }
+    }
+
+    var cornerRadius: CGFloat {
```
Comment (Jasonvdb) [inline MobileWallet/UIElements/EmoticonView.swift:79]: is this needed? Can you not just reference the above value instead?

File: MobileWallet/UIElements/TransactionDetailNavigationBar.swift [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+*/
+
+import UIKit
+
+protocol TransactionDetailNavigationBarProtocol: class {
+    var title: String? { get set }
+    var subtitle: String? { get set }
+}
+
+class TransactionDetailNavigationBar: UIView, TransactionDetailNavigationBarProtocol {
```
Comment (Jasonvdb) [inline MobileWallet/UIElements/TransactionDetailNavigationBar.swift:48]: Is the reason for implementing this custom navbar just for the subtitle label? I think maybe if we can rename it to something more generic as we'll need to use this in a lot of places. Would rather not have to do nav bar updates in multiple components

File: MobileWallet/UIElements/EmoticonView.swift (hunk 1 above)
Comment (kukabi) [reply MobileWallet/UIElements/EmoticonView.swift:71]: @SerhiiShovkoplias yeah please go through [this one](https://swift.org/documentation/api-design-guidelines).

File: MobileWallet/UIElements/TransactionDetailNavigationBar.swift (hunk 3 above)
Comment (SerhiiShovkoplias) [reply MobileWallet/UIElements/TransactionDetailNavigationBar.swift:48]: Not only for subtitle. Also for correct place for back button. On mockups we can see specific location for it. And the most important reason for custom nav bar it is possibility to animate it with our rules. But i can agree with you about naming this class and reusable possibility it.

Comment (Jasonvdb) [reply MobileWallet/UIElements/TransactionDetailNavigationBar.swift:48]: Ok great, yes I think just rename so we can start using it in other places as well

**Pull request merged**